In [1]:
%%capture
%load_ext sql
%sql sqlite:///chinook.db

'Connected: None@chinook.db'

## Check table

In [11]:
%%sql
SELECT name, type
    FROM sqlite_master
    WHERE type IN ("table", "view");

Done.


[('album', 'table'),
 ('artist', 'table'),
 ('customer', 'table'),
 ('employee', 'table'),
 ('genre', 'table'),
 ('invoice', 'table'),
 ('invoice_line', 'table'),
 ('media_type', 'table'),
 ('playlist', 'table'),
 ('playlist_track', 'table'),
 ('track', 'table')]

## A query to find out which genres sell the most tracks in the USA
* USA Genre name, number of tracks 
* each invoice contains multiple invoice_line, each invoice_line refer to a track

In [77]:
%%sql 
WITH invoice_usa AS 
    (SELECT invoice_id FROM invoice i WHERE i.billing_country = 'USA'),
     track_information AS 
    (SELECT iu.invoice_id, g.name, il.invoice_line_id
        FROM track t
        INNER JOIN genre g ON t.genre_id = g.genre_id
        INNER JOIN invoice_line il ON il.track_id = t.track_id
        INNER JOIN invoice_usa iu ON iu.invoice_id = il.invoice_id)

SELECT ti.name AS genre_name, COUNT(ti.name) AS number_sold, 
        ROUND(CAST(COUNT(ti.name) AS Float) / (SELECT COUNT(*) FROM track_information) * 100,1)
        AS sold_percentage
FROM track_information ti
GROUP BY 1
ORDER BY 2 DESC

Done.


[('Rock', 561, 53.4),
 ('Alternative & Punk', 130, 12.4),
 ('Metal', 124, 11.8),
 ('R&B/Soul', 53, 5.0),
 ('Blues', 36, 3.4),
 ('Alternative', 35, 3.3),
 ('Latin', 22, 2.1),
 ('Pop', 22, 2.1),
 ('Hip Hop/Rap', 20, 1.9),
 ('Jazz', 14, 1.3),
 ('Easy Listening', 13, 1.2),
 ('Reggae', 6, 0.6),
 ('Electronica/Dance', 5, 0.5),
 ('Classical', 4, 0.4),
 ('Heavy Metal', 3, 0.3),
 ('Soundtrack', 2, 0.2),
 ('TV Shows', 1, 0.1)]

* Based on the tracks sold in USA, 1.9% sold tracks belongs to Hip-Hop, Pop is 2.1% and Blues is 3.4, while Punks is 12.4%, so of those four artist, Red Tone which is Punk, Slim Jim Bites with genre Blues and Meteor and the Girls with genre Pop are recommended to be purchased since those three genres rank higher comparing to Hip-Hop.

## Analyze the purchases of customers belonging to each employee to see if any sales support agent is performing either better or worse than the others.
* Write a query that finds the total dollar amount of sales assigned to each sales support agent within the company. Add aby extra attributes for the employee which are relevant to the analysis

In [89]:
%%sql
WITH customer_invoice AS
    (SELECT c.customer_id customer_id, c.support_rep_id, SUM(i.total) total
     FROM customer c
     LEFT JOIN invoice i ON c.customer_id = i.customer_id
     GROUP BY 1)
SELECT 
    e.first_name||" "||e.last_name agent_name,
    e.title,
    e.hire_date,
    e.country,
    SUM(total) total
FROM (SELECT * FROM employee WHERE title = 'Sales Support Agent') e
LEFT JOIN customer_invoice ci ON e.employee_id = ci.support_rep_id
GROUP BY employee_id
ORDER BY 5 DESC

Done.


[('Jane Peacock', 'Sales Support Agent', '2017-04-01 00:00:00', 'Canada', 1731.51),
 ('Margaret Park', 'Sales Support Agent', '2017-05-03 00:00:00', 'Canada', 1584.0),
 ('Steve Johnson', 'Sales Support Agent', '2017-10-17 00:00:00', 'Canada', 1393.92)]

* As common sense, the earlier the agent is hired, the more dollar amount of sales assigned to that agent.

## Analyze the sales data for customers from each different country.
* A query that collates data on purchases from different countries

In [117]:
%%sql
CREATE VIEW customer_invoice AS
    SELECT  customer_id, 
            SUM(total) customer_total, 
            COUNT(*) customer_order_number
     FROM invoice
     GROUP BY 1;
CREATE VIEW country_sales AS
    SELECT 
    c.country, 
    COUNT(ci.customer_id) customer_number,
    SUM(ci.customer_total) total_sales,
    AVG(ci.customer_total) avg_sale_per_customer,
    SUM(ci.customer_order_number) / COUNT(ci.customer_id) avg_order_value
FROM customer c
INNER JOIN customer_invoice ci ON c.customer_id = ci.customer_id
GROUP BY c.country
ORDER BY 3 DESC;

Done.
Done.


[]

In [123]:
%%sql
SELECT cs.country, 
    customer_number, 
    total_sales, 
    avg_sale_per_customer,
    avg_order_value
FROM 
    (
    SELECT 
        cs.*,
        CASE 
            WHEN cs.customer_number = 1 THEN 1
            ELSE 0
        END AS sort
        FROM country_sales cs) cs
ORDER BY sort ASC

Done.


[('USA', 13, 1040.49, 80.03769230769231, 10),
 ('Canada', 8, 535.59, 66.94875, 9),
 ('Brazil', 5, 427.67999999999995, 85.53599999999999, 12),
 ('France', 5, 389.07, 77.814, 10),
 ('Germany', 4, 334.62, 83.655, 10),
 ('Czech Republic', 2, 273.24, 136.62, 15),
 ('United Kingdom', 3, 245.51999999999998, 81.83999999999999, 9),
 ('Portugal', 2, 185.13, 92.565, 14),
 ('India', 2, 183.14999999999998, 91.57499999999999, 10),
 ('Ireland', 1, 114.83999999999997, 114.83999999999997, 13),
 ('Spain', 1, 98.01, 98.01, 11),
 ('Chile', 1, 97.02000000000001, 97.02000000000001, 13),
 ('Australia', 1, 81.18, 81.18, 10),
 ('Finland', 1, 79.2, 79.2, 11),
 ('Hungary', 1, 78.21, 78.21, 10),
 ('Poland', 1, 76.22999999999999, 76.22999999999999, 10),
 ('Sweden', 1, 75.24, 75.24, 10),
 ('Norway', 1, 72.27000000000001, 72.27000000000001, 9),
 ('Austria', 1, 69.3, 69.3, 9),
 ('Netherlands', 1, 65.34, 65.34, 10),
 ('Belgium', 1, 60.38999999999999, 60.38999999999999, 7),
 ('Italy', 1, 50.49, 50.49, 9),
 ('Argentina', 1, 39.6, 39.6, 5),
 ('Denmark', 1, 37.61999999999999, 37.61999999999999, 10)]

## A query that categorizes each invoice as either an album purchase or not, and calculates the Number of invoices and Percentage of invoices
* Find out what percentage of purchases are individual tracks vs whole albums.
* Have to identify whether each invoice has all the tracks from an album.
* Get the list of tracks from an invoice and comparing it to the list of tracks from an album.
* The album to compare the purchase can be found by lookingup the album that one of the purchased tracks belongs to. One track belonging to a album is enough to find the album.
* Add a column to each invoice, to indicate it either an album purchase or not
* Calculate number and percentage of invoices belonging to each group

In [184]:
%%sql
WITH album_track AS
    (SELECT 
        a.album_id, t.track_id
     FROM album a
     LEFT JOIN track t ON a.album_id = t.album_id),
    invoice_track AS
    (SELECT
        invoice_id, track_id
     FROM invoice_line
     ORDER BY 1)
SELECT invoice_id,
    CASE
        WHEN (SELECT COUNT(track_id) FROM invoice_track it WHERE it.invoice_id = invoice_id) >10 THEN 1
        ELSE 0
        END as sort
FROM invoice
--SELECT track_id FROM invoice_track WHERE invoice_id = 
--SELECT track_id FROM album_track WHERE album_id = 


Done.


[(16, 1),
 (77, 1),
 (149, 1),
 (153, 1),
 (182, 1),
 (184, 1),
 (223, 1),
 (270, 1),
 (296, 1),
 (442, 1),
 (464, 1),
 (536, 1),
 (544, 1),
 (69, 1),
 (131, 1),
 (134, 1),
 (162, 1),
 (210, 1),
 (222, 1),
 (343, 1),
 (507, 1),
 (549, 1),
 (557, 1),
 (592, 1),
 (31, 1),
 (71, 1),
 (125, 1),
 (248, 1),
 (262, 1),
 (342, 1),
 (344, 1),
 (419, 1),
 (517, 1),
 (24, 1),
 (46, 1),
 (81, 1),
 (104, 1),
 (205, 1),
 (208, 1),
 (272, 1),
 (292, 1),
 (474, 1),
 (78, 1),
 (119, 1),
 (173, 1),
 (235, 1),
 (341, 1),
 (361, 1),
 (376, 1),
 (388, 1),
 (416, 1),
 (429, 1),
 (467, 1),
 (477, 1),
 (520, 1),
 (526, 1),
 (560, 1),
 (563, 1),
 (582, 1),
 (588, 1),
 (112, 1),
 (116, 1),
 (163, 1),
 (243, 1),
 (275, 1),
 (280, 1),
 (336, 1),
 (387, 1),
 (405, 1),
 (476, 1),
 (504, 1),
 (579, 1),
 (123, 1),
 (225, 1),
 (251, 1),
 (337, 1),
 (410, 1),
 (462, 1),
 (472, 1),
 (488, 1),
 (553, 1),
 (97, 1),
 (102, 1),
 (124, 1),
 (253, 1),
 (306, 1),
 (311, 1),
 (431, 1),
 (21, 1),
 (58, 1),
 (62, 1),
 (122, 1),
 (178, 1),
 (267, 1),
 (329, 1),
 (354, 1),
 (411, 1),
 (470, 1),
 (19, 1),
 (49, 1),
 (156, 1),
 (240, 1),
 (260, 1),
 (315, 1),
 (328, 1),
 (352, 1),
 (406, 1),
 (482, 1),
 (495, 1),
 (534, 1),
 (135, 1),
 (186, 1),
 (206, 1),
 (316, 1),
 (392, 1),
 (439, 1),
 (461, 1),
 (475, 1),
 (501, 1),
 (533, 1),
 (28, 1),
 (114, 1),
 (152, 1),
 (249, 1),
 (298, 1),
 (333, 1),
 (412, 1),
 (447, 1),
 (537, 1),
 (587, 1),
 (593, 1),
 (73, 1),
 (80, 1),
 (111, 1),
 (113, 1),
 (130, 1),
 (188, 1),
 (189, 1),
 (198, 1),
 (256, 1),
 (357, 1),
 (456, 1),
 (498, 1),
 (521, 1),
 (532, 1),
 (597, 1),
 (132, 1),
 (233, 1),
 (293, 1),
 (321, 1),
 (370, 1),
 (371, 1),
 (425, 1),
 (469, 1),
 (530, 1),
 (609, 1),
 (60, 1),
 (110, 1),
 (142, 1),
 (229, 1),
 (483, 1),
 (564, 1),
 (567, 1),
 (575, 1),
 (606, 1),
 (66, 1),
 (75, 1),
 (257, 1),
 (268, 1),
 (289, 1),
 (389, 1),
 (551, 1),
 (591, 1),
 (98, 1),
 (128, 1),
 (129, 1),
 (166, 1),
 (207, 1),
 (288, 1),
 (290, 1),
 (393, 1),
 (436, 1),
 (499, 1),
 (543, 1),
 (558, 1),
 (1, 1),
 (4, 1),
 (9, 1),
 (50, 1),
 (187, 1),
 (372, 1),
 (448, 1),
 (486, 1),
 (42, 1),
 (51, 1),
 (52, 1),
 (57, 1),
 (147, 1),
 (231, 1),
 (358, 1),
 (490, 1),
 (505, 1),
 (17, 1),
 (101, 1),
 (159, 1),
 (199, 1),
 (204, 1),
 (219, 1),
 (245, 1),
 (282, 1),
 (420, 1),
 (443, 1),
 (460, 1),
 (613, 1),
 (44, 1),
 (89, 1),
 (167, 1),
 (169, 1),
 (185, 1),
 (322, 1),
 (423, 1),
 (465, 1),
 (496, 1),
 (556, 1),
 (602, 1),
 (43, 1),
 (83, 1),
 (109, 1),
 (150, 1),
 (276, 1),
 (278, 1),
 (297, 1),
 (310, 1),
 (330, 1),
 (518, 1),
 (522, 1),
 (583, 1),
 (139, 1),
 (250, 1),
 (287, 1),
 (323, 1),
 (383, 1),
 (438, 1),
 (515, 1),
 (516, 1),
 (561, 1),
 (595, 1),
 (18, 1),
 (161, 1),
 (200, 1),
 (395, 1),
 (417, 1),
 (511, 1),
 (529, 1),
 (580, 1),
 (14, 1),
 (143, 1),
 (179, 1),
 (202, 1),
 (300, 1),
 (331, 1),
 (468, 1),
 (510, 1),
 (514, 1),
 (578, 1),
 (54, 1),
 (74, 1),
 (215, 1),
 (274, 1),
 (307, 1),
 (364, 1),
 (366, 1),
 (386, 1),
 (434, 1),
 (494, 1),
 (598, 1),
 (603, 1),
 (5, 1),
 (195, 1),
 (209, 1),
 (217, 1),
 (247, 1),
 (286, 1),
 (487, 1),
 (562, 1),
 (573, 1),
 (118, 1),
 (145, 1),
 (192, 1),
 (303, 1),
 (353, 1),
 (365, 1),
 (377, 1),
 (455, 1),
 (478, 1),
 (577, 1),
 (120, 1),
 (332, 1),
 (362, 1),
 (600, 1),
 (2, 1),
 (32, 1),
 (87, 1),
 (108, 1),
 (299, 1),
 (313, 1),
 (391, 1),
 (466, 1),
 (525, 1),
 (548, 1),
 (552, 1),
 (554, 1),
 (584, 1),
 (6, 1),
 (10, 1),
 (63, 1),
 (70, 1),
 (88, 1),
 (105, 1),
 (394, 1),
 (404, 1),
 (453, 1),
 (538, 1),
 (594, 1),
 (92, 1),
 (107, 1),
 (126, 1),
 (432, 1),
 (519, 1),
 (546, 1),
 (559, 1),
 (581, 1),
 (34, 1),
 (55, 1),
 (85, 1),
 (136, 1),
 (138, 1),
 (144, 1),
 (172, 1),
 (295, 1),
 (325, 1),
 (433, 1),
 (545, 1),
 (612, 1),
 (35, 1),
 (56, 1),
 (137, 1),
 (194, 1),
 (224, 1),
 (301, 1),
 (390, 1),
 (408, 1),
 (449, 1),
 (463, 1),
 (497, 1),
 (542, 1),
 (572, 1),
 (13, 1),
 (22, 1),
 (27, 1),
 (127, 1),
 (133, 1),
 (151, 1),
 (221, 1),
 (236, 1),
 (291, 1),
 (348, 1),
 (368, 1),
 (375, 1),
 (42

In [140]:
%%sql
SELECT
        invoice_id, track_id
     FROM invoice_line
    ORDER BY 1

Done.


[(1, 1158),
 (1, 1159),
 (1, 1160),
 (1, 1161),
 (1, 1162),
 (1, 1163),
 (1, 1164),
 (1, 1165),
 (1, 1166),
 (1, 1167),
 (1, 1168),
 (1, 1169),
 (1, 1170),
 (1, 1171),
 (1, 1172),
 (1, 1173),
 (2, 3476),
 (2, 482),
 (2, 2701),
 (2, 1641),
 (2, 1119),
 (2, 2324),
 (2, 201),
 (2, 819),
 (2, 392),
 (2, 2651),
 (3, 2516),
 (3, 2646),
 (4, 3448),
 (4, 2560),
 (4, 3336),
 (4, 829),
 (4, 1872),
 (4, 748),
 (4, 1778),
 (4, 2514),
 (5, 1986),
 (5, 1987),
 (5, 1988),
 (5, 1989),
 (5, 1990),
 (5, 1991),
 (5, 1992),
 (5, 1993),
 (5, 1994),
 (5, 1995),
 (5, 1996),
 (5, 1997),
 (5, 1998),
 (5, 1999),
 (5, 2000),
 (5, 2001),
 (5, 2002),
 (6, 1045),
 (6, 30),
 (7, 3477),
 (7, 3439),
 (7, 1102),
 (7, 711),
 (7, 1988),
 (7, 3024),
 (7, 834),
 (7, 2596),
 (7, 1343),
 (7, 2644),
 (7, 42),
 (8, 2231),
 (8, 2509),
 (8, 81),
 (8, 1758),
 (8, 1455),
 (8, 2946),
 (8, 574),
 (8, 1752),
 (8, 2263),
 (8, 1560),
 (9, 206),
 (9, 3456),
 (9, 1041),
 (9, 1279),
 (9, 2098),
 (9, 3399),
 (9, 1983),
 (9, 993),
 (9, 196),
 (10, 2663),
 (10, 3290),
 (11, 2597),
 (11, 2203),
 (11, 2420),
 (11, 2643),
 (11, 3072),
 (11, 1138),
 (11, 2391),
 (11, 2257),
 (11, 610),
 (11, 1419),
 (11, 3452),
 (12, 1463),
 (12, 690),
 (12, 2663),
 (12, 92),
 (13, 2553),
 (14, 541),
 (15, 3050),
 (15, 1489),
 (15, 807),
 (15, 3336),
 (16, 2160),
 (16, 875),
 (16, 83),
 (16, 16),
 (16, 1756),
 (16, 3460),
 (16, 93),
 (16, 24),
 (16, 21),
 (17, 2400),
 (17, 1610),
 (17, 1973),
 (17, 2553),
 (17, 59),
 (17, 2270),
 (17, 55),
 (17, 2564),
 (17, 2171),
 (17, 2522),
 (17, 1419),
 (18, 1027),
 (18, 3142),
 (18, 1583),
 (18, 1752),
 (18, 2736),
 (19, 105),
 (19, 2669),
 (19, 1784),
 (19, 3460),
 (19, 2640),
 (19, 1310),
 (19, 2510),
 (20, 905),
 (20, 102),
 (20, 3336),
 (20, 1132),
 (20, 337),
 (20, 60),
 (20, 3357),
 (20, 3225),
 (20, 3297),
 (20, 3297),
 (21, 1461),
 (21, 2556),
 (21, 1434),
 (21, 3135),
 (21, 1473),
 (21, 1502),
 (21, 13),
 (21, 2971),
 (21, 2561),
 (22, 1000),
 (22, 383),
 (23, 1),
 (23, 6),
 (23, 7),
 (23, 8),
 (23, 9),
 (23, 10),
 (23, 11),
 (23, 12),
 (23, 13),
 (23, 14),
 (24, 1146),
 (24, 1147),
 (24, 1148),
 (24, 1149),
 (24, 1150),
 (24, 1151),
 (24, 1152),
 (24, 1153),
 (24, 1154),
 (24, 1155),
 (24, 1156),
 (24, 1157),
 (25, 46),
 (25, 948),
 (25, 2185),
 (25, 3049),
 (25, 1167),
 (25, 32),
 (25, 1800),
 (25, 1477),
 (26, 2670),
 (26, 85),
 (27, 3439),
 (27, 3014),
 (27, 1280),
 (27, 1496),
 (27, 2229),
 (27, 1445),
 (27, 2230),
 (27, 3105),
 (28, 36),
 (28, 3456),
 (28, 932),
 (29, 1025),
 (29, 2462),
 (29, 3341),
 (29, 1626),
 (29, 2395),
 (29, 479),
 (30, 919),
 (30, 3461),
 (30, 33),
 (30, 2341),
 (30, 1139),
 (30, 3109),
 (30, 2013),
 (30, 990),
 (30, 2413),
 (30, 2396),
 (30, 3464),
 (31, 2731),
 (31, 2732),
 (31, 2733),
 (31, 2734),
 (31, 2735),
 (31, 2736),
 (31, 2737),
 (31, 2738),
 (31, 2739),
 (31, 2740),
 (31, 2741),
 (31, 2742),
 (31, 2743),
 (31, 2744),
 (31, 2745),
 (31, 2746),
 (31, 2747),
 (31, 2748),
 (31, 2749),
 (31, 2750),
 (32, 38),
 (32, 39),
 (32, 40),
 (32, 41),
 (32, 42),
 (32, 43),
 (32, 44),
 (32, 45),
 (32, 46),
 (32, 47),
 (32, 48),
 (32, 49),
 (32, 50),
 (33, 3307),
 (33, 2736),
 (33, 2542),
 (33, 2697),
 (34, 610),
 (35, 2377),
 (35, 80),
 (35, 836),
 (35, 6),
 (35, 3475),
 (36, 1582),
 (36, 968),
 (36, 2233),
 (36, 153),
 (36, 1493),
 (36, 2656),
 (36, 1168),
 (37, 3458),
 (37, 1428),
 (37, 3329),
 (37, 698),
 (37, 1014),
 (37, 3336),
 (37, 3297),
 (37, 1166),
 (37, 3392),
 (37, 1763),
 (38, 2960),
 (38, 154),
 (38, 3272),
 (38, 2557),
 (38, 1566),
 (38, 1472),
 (38, 3409),
 (39, 2308),
 (39, 1669),
 (39, 2415),
 (39, 3142),
 (39, 3366),
 (39, 184),
 (39, 2519),
 (39, 3424),
 (40, 2382),
 (40, 2654),
 (40, 2556),
 (40, 108),
 (40, 553),
 (40, 2737),
 (40, 686),
 (40, 1626),
 (40, 3460),
 (40, 2237),
 (41, 153),
 (41, 3457),
 (41, 2511),
 (41, 2698),
 (41, 2945),
 (41, 1947),
 (41, 3336),
 (41, 773),
 (42, 2646),
 (42, 3074),
 (42, 711),
 (42, 13),
 (42, 1047),
 (42, 699),
 (42, 580),
 (42, 897),
 (42, 153),
 (43, 918),
 (43, 2342),

In [165]:
%%sql
SELECT e.*,
        CASE 
            WHEN e.first_name = 'And' THEN 1
            ELSE 0
            END AS aaa
FROM employee e

Done.


[(1, 'Adams', 'Andrew', 'General Manager', None, '1962-02-18 00:00:00', '2016-08-14 00:00:00', '11120 Jasper Ave NW', 'Edmonton', 'AB', 'Canada', 'T5K 2N1', '+1 (780) 428-9482', '+1 (780) 428-3457', 'andrew@chinookcorp.com', 0),
 (2, 'Edwards', 'Nancy', 'Sales Manager', 1, '1958-12-08 00:00:00', '2016-05-01 00:00:00', '825 8 Ave SW', 'Calgary', 'AB', 'Canada', 'T2P 2T3', '+1 (403) 262-3443', '+1 (403) 262-3322', 'nancy@chinookcorp.com', 0),
 (3, 'Peacock', 'Jane', 'Sales Support Agent', 2, '1973-08-29 00:00:00', '2017-04-01 00:00:00', '1111 6 Ave SW', 'Calgary', 'AB', 'Canada', 'T2P 5M5', '+1 (403) 262-3443', '+1 (403) 262-6712', 'jane@chinookcorp.com', 0),
 (4, 'Park', 'Margaret', 'Sales Support Agent', 2, '1947-09-19 00:00:00', '2017-05-03 00:00:00', '683 10 Street SW', 'Calgary', 'AB', 'Canada', 'T2P 5G3', '+1 (403) 263-4423', '+1 (403) 263-4289', 'margaret@chinookcorp.com', 0),
 (5, 'Johnson', 'Steve', 'Sales Support Agent', 2, '1965-03-03 00:00:00', '2017-10-17 00:00:00', '7727B 41 Ave', 'Calgary', 'AB', 'Canada', 'T3B 1Y7', '1 (780) 836-9987', '1 (780) 836-9543', 'steve@chinookcorp.com', 0),
 (6, 'Mitchell', 'Michael', 'IT Manager', 1, '1973-07-01 00:00:00', '2016-10-17 00:00:00', '5827 Bowness Road NW', 'Calgary', 'AB', 'Canada', 'T3B 0C5', '+1 (403) 246-9887', '+1 (403) 246-9899', 'michael@chinookcorp.com', 0),
 (7, 'King', 'Robert', 'IT Staff', 6, '1970-05-29 00:00:00', '2017-01-02 00:00:00', '590 Columbia Boulevard West', 'Lethbridge', 'AB', 'Canada', 'T1K 5N8', '+1 (403) 456-9986', '+1 (403) 456-8485', 'robert@chinookcorp.com', 0),
 (8, 'Callahan', 'Laura', 'IT Staff', 6, '1968-01-09 00:00:00', '2017-03-04 00:00:00', '923 7 ST NW', 'Lethbridge', 'AB', 'Canada', 'T1H 1Y8', '+1 (403) 467-3351', '+1 (403) 467-8772', 'laura@chinookcorp.com', 0)]